### PR TITLE
Configure public host to correctly redirect from server

### PR DESCRIPTION
**Pull request by robinandeer**
## What?

Configure public host name to be able to correctly redirect from the server.

## Why?

We do redirect after performing access token exchange. It was redirecting to the wrong host name.

**Ticket(s): []**

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* master
  * **PR #1228** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/web-onboarding/1228" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/web-onboarding/1228?utm_source=stack-comment).

File: src/server/exchangeToken.ts [hunk 1]
```diff
@@ -14,12 +14,15 @@ export const exchangeTokenMiddleware: Router.IMiddleware<
...
+  const logger = ctx.state.getLogger('exchangeTokenMiddleware')
+
   const nextPath = ctx.URL.searchParams.get(SearchParams.Next)
   if (!nextPath) {
     throw new Error(`Search param ${SearchParams.Next} is required`)
   }
-  // Safe way of building redirect URL.  We don't want to redirect to external site
+  // Safe way of building redirect URL. We don't want to redirect to external site
   const nextUrl = new URL('', ctx.URL)
+  nextUrl.host = process.env.HOST_URL || nextUrl.host
```
Comment (alebedev) [inline src/server/exchangeToken.ts:25]: We cannot use any VERCEL_* params for this, right?

**Review by alebedev — APPROVED** [review]
Looks good! 

If we manage our preview domains in Vercel, would it prevent such problems in the future?

File: src/server/exchangeToken.ts (hunk 1 above)
Comment (robinandeer) [reply src/server/exchangeToken.ts:25]: This is deployed on Heroku 🤔

**Comment by robinandeer** [comment]
@alebedev we don't have any plans to migrate this to Vercel - so no.

Our public-facing domain is managed outside of Heroku so I believe we need to pass this in

**Pull request merged**